.github/workflows/archive-extension-source-code.yml

```diff
@@ -0,0 +1,67 @@
+name: Archive Extension Build And Source Code
+
+on:
+  push:
+    tags:
+      - coil-extension@*
+
+jobs:
+  save-mono-repo-source-code:
+    runs-on: ubuntu-latest
+   
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
+      - name: Step 1 - Use the Upload Artifact GitHub Action to upload the mono repo
+        uses: actions/upload-artifact@v2
+        with: 
+          name: source-${{env.RELEASE_VERSION}}
+          path: ${{ github.workspace }}
```

Comment:
If I'm not mistaken, this is still saving the .git folder, which means there will be still problems uploading to Mozilla Addons